src/core/mac/mac.cpp

```diff
@@ -2050,8 +2210,130 @@ uint8_t Mac::GetTimeIeOffset(const Frame &aFrame)
 }
 #endif
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+void Mac::SetCslChannel(uint8_t aChannel)
+{
+    VerifyOrExit(GetCslChannel() != aChannel, OT_NOOP);
+
+    mSubMac.SetCslChannel(aChannel);
+
+    if (mSubMac.IsCslStarted())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+exit:
+    return;
+}
+
+otError Mac::SetCslPeriod(uint16_t aPeriod)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(GetCslPeriod() != aPeriod, OT_NOOP);
+    VerifyOrExit(!GetRxOnWhenIdle(), error = OT_ERROR_INVALID_STATE);
+
+    mSubMac.SetCslPeriod(aPeriod);
+
+    if (aPeriod == 0)
+    {
+        StopCsl();
+    }
+    else if (mSubMac.IsCslStarted())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+    else
+    {
+        IgnoreError(StartCsl());
+    }
+
```

Comment:
Need to call `UpdateIdlMode()` from here. 

Since the  change in CSL period may start/stop CSL operation. If `Mac` is already in idle mode/operation we would need to udpate `SubMac` state accordingly.